report/renderer/details-renderer.js

```diff
@@ -376,7 +380,96 @@ export class DetailsRenderer {
   }
 
   /**
-   * @param {{headings: TableColumnHeading[], items: TableItem[]}} details
+   * Adorn a table row element with entity chips based on [data-entity] attribute.
+   * @param {HTMLTableRowElement} rowEl
+   */
+  _adornTableRowWithEntityChips(rowEl) {
+    const entityName = rowEl.dataset.entity;
+    if (!this._entities || !entityName) return;
+    const entityIndex = this._entities.entityIndexByName[entityName];
```

Comment:
Is `entityIndexByName` really necessary here? Couldn't we just do `this._entities.list.find(e => e.name === entityName)`?

Reply:
That's an unnecessary `O(n)` scan operation per row rendered, and hence we have the lookup tables.

Reply:
How many entities is any page realistically going to have? Adding an extra data structure to the lhr to reduce time complexity seems like splitting hairs IMO.

Reply:
I don't think we have the data yet to count the entities that would be detected for an average case yet. Once we ship entity-classification, we should be able to query. FWIW, a page like theverge.com homepage seems to have anywhere between 50-100 entities detected. We could do the helper function.